src/S3/Result/ListObjectsOutput.php

```diff
@@ -105,6 +105,22 @@ public function getIsTruncated(): ?bool
         return $this->IsTruncated;
     }
 
+    /**
+     * Iterates over Contents and CommonPrefixes.
+     *
+     * @return ObjectAndCommonPrefix[]
+     */
+    public function getIterator(): \Traversable
+    {
+        $this->initialize();
+
+        return (function () {
+            foreach ($this->Contents as $index => $item) {
+                yield new ObjectAndCommonPrefix($item, $this->CommonPrefixes[$index] ?? null);
```

Comment:
Is  the `$this->Contents` array always longer than `$this->CommonPrefixes`?

Reply:
Hmmmm... I totaly missunderstood how it works. After investigating: . In fact when `result_key` contains several keys, they are not related..
For instance, when you call `s3::ListObject`, you have 2 list: `contents` that contains the objects and `commonPrefixes` that contains list of "folders". Mixing theme does not make sens. 

I suggest to either iterrating over Content then CommonPrefixes and the user have to deal with `if($item instanceOf AwsObject) ...`
Or to provide 2 methods: getContentAsTraversable and getCommonPrefixesAsTraversable


WDYT?

Reply:
Let’s iterate over them both. That is how flysystem does. 

If you really want to have only one object type, you still can... (sure, but without pagination)